manifests/00-custom-resource-definition.yaml

```diff
@@ -292,10 +292,41 @@ spec:
                                 - Local
                                 type: string
                             type: object
+                          ibm:
+                            description: "ibm provides configuration settings that
+                              are specific to IBM Cloud load balancers. \n If empty,
+                              defaults will be applied. See specific ibm fields for
+                              details about their defaults."
+                            properties:
+                              enableFeatures:
+                                description: "enableFeatures can be used to enable
+                                  features on IBMCloud loadbalancers \n See \"service.kubernetes.io/ibm-load-balancer-cloud-provider-enable-features\"
+                                  at https://cloud.ibm.com/docs/containers?topic=containers-vpc-lbaas"
+                                properties:
+                                  proxyProtocol:
+                                    description: "Enable PROXY protocol on the loadbalancer.
+                                      The load balancer passes client connection information,
+                                      including the client IP address, the proxy server
+                                      IP address, and both port numbers, in request
+                                      headers to your back-end app. \n See \"service.kubernetes.io/ibm-load-balancer-cloud-provider-enable-features:
+                                      \"proxy-protocol\"\" at https://cloud.ibm.com/docs/containers?topic=containers-vpc-lbaas"
```

Comment:
This description should mention that PROXY protocol cannot be toggled on an ingresscontroller after it has been created; toggling will require deleting and recreating the ingresscontroller.  

If you actually want toggling PROXY protocol to be possible without deleting and recreating the ingresscontroller, then `setDefaultPublishingStrategy` needs some logic to check for updates to `spec.endpointPublishingStrategy.loadBalancer.providerParameters.ibm.enableFeatures.proxyProtocol`.  However, toggling the setting would most likely cause brief disruption as the LB and router deployment would be updated asynchronously, and the router drops traffic during any period of time during which the configuration is inconsistent (that is, the LB is sending PROXY and HAProxy is expecting TCP or vice versa).  

Would it be feasible to use the `IngressControllerProtocol` type for this field?  That would make the `spec.endpointPublishingStrategy.loadBalancer.providerParameters.ibm.enableFeatures.proxyProtocol` API field more consistent with the `spec.endpointPublishingStrategy.hostNetwork.protocol`, `spec.endpointPublishingStrategy.private.protocol`, and `spec.endpointPublishingStrategy.nodePort.protocol` fields.  You could also re-use some of the wording from the godoc for those fields.

Reply:
- Updated the variable in `openshift/api`: Renamed it to`spec.endpointPublishingStrategy.loadBalancer.providerParameters.ibm.protocol` and changed the type from `bool` to `IngressControllerProtocol`. 
- Added logic to `setDefaultPublishingStrategy` so routers get updated after setting `spec.endpointPublishingStrategy.loadBalancer.providerParameters.ibm.protocol`.

One thing that doesn't seem to work is automatic loadbalancer updates. On IBMCloud the proxy protocol annotation has to be added to the loadbalancer during creation time for it to work. This means that the ingress operator would have to delete the loadbalancer and recreate it. We can also just document and ask the users to delete the loadbalancer service manually and ingress operator will recreate it for them.